examples/copying-characters-into-array-with-foreach.flix

```diff
@@ -1,25 +0,0 @@
-// Easily iterate over arrays with foreach-loops.
```

Comment:
I would like to keep these examples; if they cannot easily be reworked, perhaps we should just comment them for now?

Reply:
I am just afraid of losing them to the sands of time :)

Reply:
I could rewrite them with `forEachWithIndex` and put a comment saying to update them when we have Streams / mutable enumerators. Would that work?

Reply:
That would be perfect :)

Reply:
I've re-instated them. This included adding `forEachWithIndex` to Iterator which is a workaround because `enumerator` is broken.